samples/js/photos/src/photo-grid.js

```diff
@@ -0,0 +1,355 @@
+    const {size} = this.state;
+    const widthScale = gridWidth / size.width;
+    const heightScale = gridHeight / size.height;
+
+    return {
+      overflow: 'hidden',
+      position: 'absolute',
+      transition: this._maybeTransition(`z-index ${transitionDelay} step-end,
+                                         transform ${transitionDelay} ease`),
+      // This is an awkward way to implement transformOrigin, but we can't use
+      // it because it doesn't work well with transitions.
+      transform: `translate3d(${-size.width / 2}px, ${-size.height / 2}px, 0)
+                  scale3d(${widthScale}, ${heightScale}, 1)
+                  translate3d(${(size.width / 2)}px, ${(size.height / 2)}px, 0)
+                  translate3d(${gridLeft / widthScale}px, ${gridTop / heightScale}px, 0)`,
+    };
+  }
+
+  _getClipFullscreenStyle(): {[key: string]: any} {
+    // We'll scale the image, not the surrounding div, which is used by the grid to clip the image.
+    const {viewport} = this.props;
+    const {size} = this.state;
+
+    // Figure out whether we should be scaling width vs height to the dimensions of the screen.
+    const widthScale = viewport.clientWidth / size.width;
+    const heightScale = viewport.clientHeight / size.height;
+    const scale = Math.min(widthScale, heightScale);
+
+    // Transform the image to the center of the screen.
+    const middleLeft = (viewport.clientWidth - size.width) / 2 - this._parentLeft;
+    const middleTop = (viewport.clientHeight - size.height) / 2 - this._parentTop;
+
+    return {
+      position: 'absolute',
+      transition: this._maybeTransition(`z-index ${transitionDelay} step-start,
+                                         transform ${transitionDelay} ease`),
+      // TODO: There appears to be a rounding error in here somewhere which causes the high-res
+      // fullscreen image to be vertically ~1px low. Perhaps scaling before translating would help?
+      transform: `translate3d(${middleLeft}px, ${middleTop}px, 0) scale3d(${scale}, ${scale}, 1)`,
+      zIndex: 1,
+    };
+  }
+
+  _getImgGridStyle(): {[key: string]: any} {
+    const {gridWidth, gridHeight} = this.props;
+    const {size} = this.state;
+    const widthScale = gridWidth / size.width;
+    const heightScale = gridHeight / size.height;
+
+    // Reverse the width scale that the outer div was scaled by, then scale the image's width by how
+    // the outer div's height was scaled (so that the proportions are correct).
+    return {
+      transition: this._maybeTransition(`transform ${transitionDelay} ease`),
+      transform: `scale3d(${heightScale / widthScale}, 1, 1)`,
+    };
+  }
+
+  _getImgFullscreenStyle(): {[key: string]: any} {
+    return {
+      transition: this._maybeTransition(`transform ${transitionDelay} ease`),
+    };
+  }
+
+  _maybeTransition(transition: string): string {
+    return this._shouldTransition === true ? transition : '';
```

Comment:
Yes, when I remember, I try to do this. Flow doesn't catch misspelled object properties when you use them as conditionals, if you do `=== true` then it will.